src/core/planetsephems/jpleph.cpp

```diff
@@ -578,20 +581,21 @@ int DLL_FUNC jpl_state(void *ephem, const double et, const int list[14],
 {
 	struct jpl_eph_data *eph = static_cast<struct jpl_eph_data *>(ephem);
 	unsigned i, j, n_intervals;
-	uint32_t nr;
 	double *buf = eph->cache;
 	double t[2];
 	const double block_loc = (et - eph->ephem_start) / eph->ephem_step;
 	bool recompute_pvsun;
 	const double aufac = 1.0 / eph->au;
 
+	mutex.lock();
```

Comment:
`QMutexLocker` would be useful here. Using it you'll no longer need to call `unlock` before every return, the destructor will to it for you.